pkgs/applications/networking/owncloud-client/default.nix

```diff
@@ -1,16 +1,32 @@
-{ lib, fetchurl, mkDerivation, cmake, extra-cmake-modules, pkg-config, qtbase, qtkeychain, sqlite, libsecret }:
+{ lib
+, fetchFromGitHub
+, mkDerivation
+, pkg-config
+, cmake
+, extra-cmake-modules
+, callPackage
+, qtbase
+, qtkeychain
+, qttools
+, sqlite
+, libsecret
+}:
 
 mkDerivation rec {
   pname = "owncloud-client";
-  version = "2.11.0.8354";
+  version = "3.2.1";
 
-  src = fetchurl {
-    url = "https://download.owncloud.com/desktop/ownCloud/stable/${version}/source/ownCloud-${version}.tar.xz";
-    sha256 = "sha256-YraWvGgeF5b1+3i5Jlk+bwvAULr7KFOSX8y0ZoPpljI=";
+  libregraph = callPackage ./libre-graph-api-cpp-qt-client.nix { };
+
+  src = fetchFromGitHub {
+    owner = "owncloud";
+    repo = "client";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-39tpvzlTy3KRxg8DzCQW2VnsaLqJ+dNQRur2TqRZytE=";
   };
 
   nativeBuildInputs = [ pkg-config cmake extra-cmake-modules ];
-  buildInputs = [ qtbase qtkeychain sqlite libsecret ];
+  buildInputs = [ qtbase qttools qtkeychain sqlite libsecret libregraph ];
```

Comment:
qttools goes to nativeBuildInputs

Reply:
I tried some fixes in https://github.com/NixOS/nixpkgs/pull/230081 but I am not sure if it's the right approach, if you could give some hints that'd be great.